cmd/performance-profile-creator/cmd/root.go

```diff
@@ -18,27 +18,202 @@ package cmd
+	PreRunE: func(cmd *cobra.Command, args []string) error {
+		mcpName := cmd.Flag("mcp-name").Value.String()
+		mustGatherDirPath := cmd.Flag("must-gather-dir-path").Value.String()
+		mcp, err := getMCPByName(mustGatherDirPath, mcpName)
+		if err != nil {
+			return fmt.Errorf("Error obtaining MachineConfigPool %s: %v", mcpName, err)
+		}
+		labelSelector := mcp.Spec.NodeSelector
+		nodes, err := getNodeList(mustGatherDirPath)
+		if err != nil {
+			return fmt.Errorf("Error obtaining Nods %s: %v", mcpName, err)
+		}
+
+		matchedNodes := make([]*v1.Node, 0)
+		for _, node := range nodes {
+			matches, _ := corev1.MatchNodeSelectorTerms(node, getNodeSelectorFromLabelSelector(labelSelector))
+			if matches {
+				log.Println("Matched Node:", node.GetName())
+				matchedNodes = append(matchedNodes, node)
+			}
+		}
+		return nil
+	},
 	Run: func(cmd *cobra.Command, args []string) {
 		profileName := cmd.Flag("profile-name").Value.String()
 		createProfile(profileName)
 	},
 }
 
+func getNodeSelectorFromLabelSelector(labelSelector *metav1.LabelSelector) *v1.NodeSelector {
+
+	matchExpressions := make([]v1.NodeSelectorRequirement, 0)
+	for key, value := range labelSelector.MatchLabels {
+		matchExpressions = append(matchExpressions, v1.NodeSelectorRequirement{
+			Key:      key,
+			Operator: v1.NodeSelectorOpIn,
+			Values:   []string{value},
+		})
+	}
+	matchFields := make([]v1.NodeSelectorRequirement, 0)
+	for _, labelSelectorRequirement := range labelSelector.MatchExpressions {
+		matchExpressions = append(matchFields, v1.NodeSelectorRequirement{
+			Key:      labelSelectorRequirement.Key,
+			Operator: v1.NodeSelectorOperator(string(labelSelectorRequirement.Operator)),
+			Values:   labelSelectorRequirement.Values,
+		})
+	}
+
+	nodeSelectorTerms := []v1.NodeSelectorTerm{
+		{
+			MatchExpressions: matchExpressions,
+			MatchFields:      matchFields,
+		},
+	}
+	nodeSelector := &v1.NodeSelector{
+		NodeSelectorTerms: nodeSelectorTerms,
+	}
+
+	return nodeSelector
+
+}
+
+func getMustGatherFullPaths(mustGatherPath string, suffix string) (string, error) {
+	// The regular expression below depends on the must gather image name. It is assumed here
+	// that the image would have "performance-addon-operator-must-gather" substring in the name.
+	paths, err := filepath.Glob(mustGatherPath + "/*performance-addon-operator-must-gather*/" + suffix)
+	// we return only one path here, we could add a validation
+	// 1) If len(paths) == 1 i.e. there is only one possible subdirectory under must gather prefix with the above prefix, we return the path
+	// 2) If len(paths) != 1 we choose one, one option could be to check the timestamp of the directory and choose the most fresh one
+	return paths[0], err
+}
+
+func getNode(path string) (*v1.Node, error) {
+	var node v1.Node
+	src, err := os.Open(path)
+	if err != nil {
+		return nil, fmt.Errorf("Error opening %q: %v", path, err)
+	}
+	defer src.Close()
+
+	dec := k8syaml.NewYAMLOrJSONDecoder(src, 1024)
```

Comment:
How do we know 1024 bytes are enough?

Reply:
1024 is the bufferSize value which determines how far into the stream the decoder will look to figure out whether this is a JSON stream (has whitespace followed by an open brace), so in this case we know that we son't have whitespaces and the yaml is present right from the beginning of the file so 1024 is a safe choice.